### PR TITLE
Bump up libraries to eliminate exploitable vulnerabilities

### PR DESCRIPTION
**Pull request by bd-samratmuk**
Libraries updated -
- json-path
- guava
- okio
- bouncycastle
- kubernetes-client

Notes on Logback 1.2.31 and [CVE-2026-1225](https://nvd.nist.gov/vuln/detail/CVE-2026-1225)

Upgrading to Logback 1.5+ (and Java 11) is scheduled for version 12.0. In the meantime, detect-docker-inspector is not vulnerable to CVE-2026-1225 for the following reasons:

- No Attack Surface: The project does not use a custom logback.xml file; Logback is managed entirely via Spring Boot auto-configuration. There is no local configuration file for an attacker to modify or exploit.
- High Barrier to Entry: Exploiting this vulnerability requires write access to the filesystem or the ability to inject JVM startup parameters (e.g., -Dlogback.configurationFile). If an attacker possesses this level of access, they could already execute far more critical attacks, such as replacing the JAR or injecting a Java agent.
- Low Impact: The CVE notes that even if a malicious class is successfully instantiated, it is "very likely to be discarded with no further ado," resulting in negligible impact on the system. 

In short, the existing security controls and Spring Boot configuration eliminate the practical risk posed by this CVE.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -83,6 +84,23 @@ repositories {
 configurations {
     integrationTestCompile.extendsFrom testCompile
     integrationTestRuntime.extendsFrom testRuntime
+
+    // Force guava upgrade to fix known vulnerability (transitive from docker-java)
+    all {
+        resolutionStrategy {
+            force 'com.google.guava:guava:33.6.0-android'
```
Comment (dterrybd) [inline build.gradle]: why not 33.6.0-jre? It seems weird to bring in an android version of the library?

File: build.gradle [hunk 2]
```diff
@@ -342,6 +360,13 @@ artifactory {
     }
 }
 
+dependencyManagement {
+    dependencies {
+        // Override json-path managed by Spring Boot BOM to fix vulnerability
+        dependency 'com.jayway.jsonpath:json-path:2.10.0'
```
Comment (dterrybd) [inline build.gradle:366]: The other two changes, specifically the one in resolutionStrategy, doesn't cover us? If you remove this something fails?

File: build.gradle [hunk 3]
```diff
@@ -9,6 +9,7 @@ buildscript {
         javaSourceCompatibility = 8
     }
     ext['logback.version'] = '1.2.13'
+    ext['json-path.version'] = '2.10.0'
```
Comment (zahidblackduck) [inline build.gradle:12]: I believe the exploitable transitive dependency is `json-smart` if I'm not wrong. I don't have access to the `detect-docker-inspector` pop scan sbom to verify this.
And, `json-smart` comes via this `json-path` from another upstream library, possibly `integration-common` (transitively brought by `integration-rest` here).
The good news is `integration-common` has been updated to `json-path:2.10.0`. So, the downstream flow of `json-smart` is now to the `2.6.0` version, which is safer.
Could you kindly try to update the `integration-rest` to it's latest version? 

Sure, another path can be `hub-imageinspector-lib` used here which uses `integration-bdio` and that relies on `integration-common` too.

We've to cover all the path to make the downstream libraries safer. I'm currently working on the `hub-imageinspector-lib` fix, which should use the latest released `integration-bdio` containing the fix. And, `integration-bdio` has been updated too to contain the latest fix.

I know this approach of fixing the vulnerabilities is a bit time consuming, but it seems to be the more preferred way as discussed with @dterrybd earlier.

**Review by zahidblackduck — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (zahidblackduck) [reply build.gradle]: I've been exploring the guava fix in `bdio` library. Let me share my findings here.

We are proposing the hub team to upgrade to `com.google.guava:guava:32.0.1-jre`

Smallest jump that fixes the CVE. Both CVE-2020-8908 and CVE-2023-2976 were fixed in `32.0.0`. Anything at or above `32.0.0` clears the scan. `33.0.0` or higher version has real breaking changes that `32.0.1` does not as per the [changelog](https://github.com/google/guava/releases?page=2#:~:text=vs.%2033.0.0%2Djre-,Changelog,-This%20version%20of) of guava.

`32.0.1` has zero breaking changes beyond the security fix itself (`Files.createTempDir()` reimplementation), which is the point of the fix. Every additional minor version you cross is another set of changelogs to audit and another set of edge cases to test for. Since `32.0.1` already clears the CVE, there is no reason to jump further.

And, regarding the `jre` vs. `android` comment by @dterrybd 

Guava publishes two flavours of every release. 
The `-jre` flavour is the correct choice for any non-android project. This is also what `bdio2` already uses (`guava:30.1.1` resolves to the `-jre` classifier in artifactory), so switching to 32.0.1-jre preserves the existing classifier. Further exploration needs to be done before switching to android flavor.

**Review by zahidblackduck — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (bd-samratmuk) [reply build.gradle]: Nice catch. I went with the android flavour to support the latest version. But I believe your proposal is more accurate with guava:32.0.1-jre. Addressed in the latest commit.

File: build.gradle (hunk 2 above)
Comment (bd-samratmuk) [reply build.gradle:366]: spring can apply extensive dependency management so this change was needed for overriding.

File: build.gradle (hunk 3 above)
Comment (bd-samratmuk) [reply build.gradle:12]: This is actually the right approach. But this project is little different like we discussed. In this project `detect-docker-inspector` - `json-path` is coming from both `integration-common` as well as `spring-boot-starter-test`. In that case Hub only shows `spring-boot-starter-test` as parent of `json-path` because `json-path` is a direct dependency of `spring-boot-starter-test`. When I set the scope prop to runtime(`--detect.gradle.included.configurations=runtimeClasspath`) then only the hub reveals `integration-common` as parent of` json-path`. Upgrading both is the correct approach here but Spring update is not possible atm so needed to override it.

**Review by zahidblackduck — APPROVED** [review]

**Review by dterrybd — APPROVED** [review]

File: build.gradle [hunk 4]
```diff
@@ -342,8 +360,15 @@ artifactory {
...
+    dependencies {
+        // Override json-path managed by Spring Boot BOM to fix vulnerability
+        dependency 'com.jayway.jsonpath:json-path:2.10.0'
+    }
+}
+
 dependencies {
-    implementation "com.blackduck.integration:integration-rest:11.1.2"
+    implementation "com.blackduck.integration:integration-rest:11.1.4"
     implementation 'com.blackduck.integration:hub-imageinspector-lib:15.0.4'
```
Comment (zahidblackduck) [inline build.gradle]: `hub-imageinspector-lib:15.0.5` has been released, which aims to fix vulnerable `json-smart` dependency. So, updating to the latest version can be considered.

Comment (bd-samratmuk) [reply build.gradle]: Updated.

**Review by zahidblackduck — APPROVED** [review]
